op-chain-ops/cmd/opbnb-upgrade/main.go

```diff
@@ -0,0 +1,172 @@
+
+	app := &cli.App{
+		Name:  "opbnb-upgrade",
+		Usage: "Build transactions useful for upgrading the opBNB",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "l1-rpc-url",
+				Value:   "http://127.0.0.1:8545",
+				Usage:   "L1 RPC URL, the chain ID will be used to determine the opBNB",
+				EnvVars: []string{"L1_RPC_URL"},
+			},
+			&cli.PathFlag{
+				Name:    "outfile",
+				Usage:   "The file to write the output to. If not specified, output is written to stdout",
+				EnvVars: []string{"OUTFILE"},
+			},
+			&cli.PathFlag{
+				Name:    "transfer_owner",
+				Usage:   "Transfer proxyAdmin contract owner to the address",
+				EnvVars: []string{"TRANSFER_OWNER"},
+			},
+			&cli.PathFlag{
+				Name:    "private_key",
+				Usage:   "Owner private key to transfer new owner",
+				EnvVars: []string{"PRIVATE_KEY"},
+			},
+			&cli.PathFlag{
+				Name:    "qa_net",
+				Usage:   "set network is qanet",
+				EnvVars: []string{"QA_NET"},
+			},
+		},
+		Action: entrypoint,
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		log.Crit("error opBNB-upgrade", "err", err)
+	}
+}
+
+// entrypoint contains the main logic of the script
+func entrypoint(ctx *cli.Context) error {
+	client, err := ethclient.Dial(ctx.String("l1-rpc-url"))
+	if err != nil {
+		return err
+	}
+
+	// Fetch the L1 chain ID to determine the superchain name
+	l1ChainID, err := client.ChainID(ctx.Context)
+	if err != nil {
+		return err
+	}
+
+	proxyAddresses := opbnb_upgrades.BscQAnetProxyContracts
+	implAddresses := opbnb_upgrades.BscQAnetImplContracts
+	if l1ChainID.Uint64() == opbnb_upgrades.BscTestnet && !ctx.IsSet("qa_net") {
+		proxyAddresses = opbnb_upgrades.BscTestnetProxyContracts
+		implAddresses = opbnb_upgrades.BscTestnetImplContracts
+		fmt.Println("upgrade bscTestnet")
+	} else if l1ChainID.Uint64() == opbnb_upgrades.BscMainnet {
+		proxyAddresses = opbnb_upgrades.BscMainnetProxyContracts
+		implAddresses = opbnb_upgrades.BscMainnetImplContracts
+		fmt.Println("upgrade bscMainnet")
+	} else {
+		fmt.Println("upgrade bscQAnet")
+	}
+
+	if ctx.IsSet("transfer_owner") {
+		if !ctx.IsSet("private_key") {
+			return errors.New("must set private key")
+		}
+		proxyAdmin, err := oldBindings.NewProxyAdmin(implAddresses["ProxyAdmin"], client)
+		if err != nil {
+			return err
+		}
+		owner, err := proxyAdmin.Owner(&bind.CallOpts{})
+		if err != nil {
+			return err
+		}
+		fmt.Printf("old proxyAdmin owner is %s\n", owner.String())
+		transferOwner := ctx.String("transfer_owner")
+		privateKeyHex := ctx.String("private_key")
+		privateKey, err := crypto.HexToECDSA(privateKeyHex)
+		if err != nil {
+			return err
+		}
+		txOpts, err := bind.NewKeyedTransactorWithChainID(privateKey, l1ChainID)
+		if err != nil {
+			return err
+		}
+		tx, err := proxyAdmin.TransferOwnership(txOpts, common.HexToAddress(transferOwner))
+		if err != nil {
+			return err
+		}
+		fmt.Printf("TransferOwnership tx hash is %s\n", tx.Hash())
+		time.Sleep(5 * time.Second)
```

Comment:
Maybe the sleep code can be replaced with checking the status of the receipt corresponding to the transaction